src/index.ts

```diff
@@ -116,19 +121,53 @@ const translate = async (
 
   const translationService = serviceMap[service];
 
-  const templateFilePath = evaluateFilePath(workingDir, dirStructure, sourceLang);
-
-  const templateFiles = loadTranslations(
-    templateFilePath,
-    fileType,
+  const templateFilePath = evaluateFilePath(
+    workingDir,
+    dirStructure,
+    sourceLang,
   );
 
+  const templateFiles = loadTranslations(templateFilePath, fileType);
+
   if (templateFiles.length === 0) {
     throw new Error(
       `The source language ${sourceLang} doesn't contain any JSON files.`,
     );
   }
 
+  let tabWidth: number;
+  if (cmdLineTabWidth === undefined) {
+    // Try detecting tabWidth from .prettierrc config file in working directory.
+    const prettierConfigPath = path.resolve('.prettierrc');
+    if (fs.existsSync(prettierConfigPath)) {
+      const prettierConfig = fs.readFileSync(prettierConfigPath, {
+        encoding: 'utf-8',
+      });
+      try {
+        const prettierJson = JSON.parse(prettierConfig);
+        if (
+          prettierJson.tabWidth &&
+          typeof prettierJson.tabWidth === 'number'
+        ) {
+          tabWidth = prettierJson.tabWidth;
+          console.log(
+            chalk.green(`Using JSON tabWidth (${tabWidth}) from .prettierrc.`),
+          );
+        }
+      } catch (error) {
+        console.error(
+          chalk.red(
+            `Found .prettierrc file, but is is not a valid JSON file. Using default indentation.`,
+          ),
+        );
+      }
+    } else {
+      tabWidth = 2;
+    }
+  }else {
+    tabWidth = Number.parseInt(cmdLineTabWidth);
```

Comment:
This would fail when the user didn't specify their own tab width. You could specify the default tab width on line 82 as a third argument, like I did for another option in line 57.